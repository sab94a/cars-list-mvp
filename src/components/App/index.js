//@flow

import React from 'react';
import { BrowserRouter, Route, Switch } from 'react-router-dom';
import type { Route as RouteType } from 'types/routes';
import Layout from 'components/Layout';

export type Props = {
    routes: Array<RouteType>
}

const App = ({ routes }: Props) => (
    <BrowserRouter>
        <Layout>
            <Switch>
                { routes.map((route, index) => (
                    <Route key={ index } {...route} />
                )) }
            </Switch>
        </Layout>
    </BrowserRouter>
);

export default App;
