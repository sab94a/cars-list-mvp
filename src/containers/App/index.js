//@flow

import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter, Route, Switch, Link } from 'react-router-dom';
import type { Route as RouteType } from 'types/routes';
import type { NavItem } from 'components/lib/types';
import Layout from 'components/Layout';
import LibLink from 'components/lib/Link';

type Props = {
    routes: Array<RouteType>,
    navigation: Array<NavItem>,
    strings: {
        [string]: string
    },
    store: any
};

// configurate my own library link
LibLink.Component = ({ href, children, ...rest }) => (
    <Link to={ href } {...rest}>{ children }</Link>
);

const App = ({ store, routes, navigation, strings: { copyright } }: Props) => (
    <Provider store={ store }>
        <BrowserRouter>
            <Layout navigation={ navigation } footerText={ copyright }>
                <Switch>
                    { routes.map((route, index) => (
                        <Route key={ index } {...route} />
                    )) }
                </Switch>
            </Layout>
        </BrowserRouter>
    </Provider>
);

export default React.memo<Props>(App);
