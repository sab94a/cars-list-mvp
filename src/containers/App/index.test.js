import React from 'react';
import { shallow  } from 'enzyme';
import toJson from 'enzyme-to-json';
import App from './'

describe('container <App />', () => {
    const store = {
        default: jest.fn(),
        subscribe: jest.fn(),
        dispatch: jest.fn(),
        getState: () => {},
    };

    const routes = [{
        path: 'path_1',
        component: 'component_1'
    }, {
        path: 'path_2',
        component: 'component_2'
    }];

    const navigation = [{
        title: 'title_1',
        href: '/test'
    }, {
        title: 'title_2',
        href: '/test_2'
    }];

    const strings = { copyright: 'copyright' };

    it('Should render correctly', () => {
        const component = shallow (
            <App store={ store } routes={ routes } navigation={ navigation } strings={ strings } />
        )

        expect(toJson(component)).toMatchSnapshot();
    });
});
