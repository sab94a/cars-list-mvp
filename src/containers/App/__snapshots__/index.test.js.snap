// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`container <App /> Should render correctly 1`] = `
<Provider
  store={
    Object {
      "default": [MockFunction],
      "dispatch": [MockFunction],
      "getState": [Function],
      "subscribe": [MockFunction],
    }
  }
>
  <BrowserRouter>
    <Layout
      footerText="copyright"
      navigation={
        Array [
          Object {
            "href": "/test",
            "title": "title_1",
          },
          Object {
            "href": "/test_2",
            "title": "title_2",
          },
        ]
      }
    >
      <Switch>
        <Route
          component="component_1"
          key="0"
          path="path_1"
        />
        <Route
          component="component_2"
          key="1"
          path="path_2"
        />
      </Switch>
    </Layout>
  </BrowserRouter>
</Provider>
`;
